src/formatDate.js

```diff
@@ -8,7 +8,28 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+  const dateParts = date.split(fromSeparator);
+
+  const dateComponents = {};
+
+  fromFormat.slice(0, 3).forEach((part, index) => {
+    dateComponents[part] = dateParts[index];
+  });
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    dateComponents['YYYY'] =
+      dateComponents['YY'] < 30
+        ? `20${dateComponents['YY']}`
+        : `19${dateComponents['YY']}`;
```

Comment:
The condition for checking the year format conversion from 'YY' to 'YYYY' is correct. However, the hardcoded '30' could be made into a named constant to clarify its meaning and purpose. Consider defining a constant like `const THRESHOLD_YEAR = 30;` at the top of your function and using it in place of the hardcoded '30' to make the code more self-documenting.